### DISCUSSION TITLE
Sharing devices

### DISCUSSION THREAD

**Discussion by JustMe-NL** [OMOTE-Community/OMOTE-Firmware#77] [General]
So for my own use I created a couple of new devices. Maybe we can create some place to share the files so others can reuse them without making the original repo to overwhelming for beginners to understand?

So far I have the commands for a LG TV, LG Soundbar, LG Blu-Ray player, Samsung Blu-Ray player (needs Samsung36 added in the hardware abstraction) and Nvidea Shield 2015 (P2571) which supports some IR commands.

**Comment by KlausMu** [comment]
Sounds great.

**Device pool**
I'm not sure where the best place for the pool of all available devices would be. Maybe a folder `devices_pool` which is not being compiled by default.

To use a device, the user would have to:

- copy the device files (*.cpp, *.h, maybe the gui files as well)  to folder "devices"
- add the `register_device_*` command to main.cpp

Do you think this is understandable and easy to use?

**IR Samsung36**
For this we need a PR. Could you please create one?

**Your device files**
Could you please also create a PR for them? For now, being at the default place is ok. There are not too many devices for now.

- please add the exact name of the device in "register_device_*" (see the other devices for an example)
- please activate only the most used commands and comment the rest. Each command takes memory, and activating all available commands  normally is not necessary.

**Comment by KlausMu** [comment]
I introduced a folder `src/devices_pool` and put all the devices not used by the demo application there.
This was needed, because the 5 new devices took about 15k of the remaining 25k of free memory.
Wiki is also updated.
Thanks for the devices you contributed!
